Library/Services/ContextStorage.php

```diff
@@ -227,10 +241,52 @@ public function getWorkflowStepList(string $workflowProcessId) : array {
      * @returns array
      */
     public function setWorkflowStepList(string $workflowProcessId, $workflowStepList) {
+        $this->di->get('logger')->debug(json_encode(func_get_args()));
         $contextHash = $this->get($workflowProcessId);
         $contextHash[self::WORKFLOW][self::WORKFLOW_STEPS] = $workflowStepList;
         $this->save($workflowProcessId, $contextHash);
     }
 
+    /**
+     * Update the given step related to the given workflow/stepcode
+     *
+     * @param string $workflowProcessId the workflow identifier
+     * @param string $stepCode          the step code
+     * @param array  $stepHash          the step data to save
+     *
+     * @returns array
+     */
+    public function updateWorkflowStep(string $workflowProcessId, $stepCode, $stepHash)
+    {
+        $this->di->get('logger')->debug(json_encode(func_get_args()));
+        $script = <<<EOT
```

Comment:
perhaps we can check if context storage adapter manage partial update.

if ($this->adpater->hasUpdatePartial())
   // use update partial. ..
else 
   // use update classic

Reply:
partial update is required, no choice.